### ISSUE TITLE
Crash on DL tab CLEAR files right-click option

### ISSUE THREAD

**Issue by DeathStalker77** [nicotine-plus/nicotine-plus#2745]
Nicotine+ version:
Nicotine+ Version: 3.3.0.dev6
GTK Version: 4.12.3
Python Version: 3.11.6 (win32)

Operating System/Distribution:
Win10 x64 Ent
22H2 19045.3570
Windows Feature Experience Pack 1000.19052.1000.0

### Describe the bug
Selected COMPLTED items to clear via right-click in DOWNLOAD tab - turned out to be all of the items queued for that user - and N+ crashed -

Type: <class 'KeyError'>
Value: 'Beaubrummel73@@cuqhu\\Echo Sharing\\Yes\\22 - 35th Anniversary Tour [Apr 2004 - Sep 2004]\\040515 [15-May-04] Lowell 2004\\2004-05-15 - Lowell 2004 - 303 Starship Trooper.flac'
Traceback:   File "D:/a/nicotine-plus/nicotine-plus/pynicotine/gtkgui/transfers.py", line 1062, in on_clear_transfer
  File "D:/a/nicotine-plus/nicotine-plus/pynicotine/gtkgui/downloads.py", line 104, in clear_selected_transfers
  File "D:/a/nicotine-plus/nicotine-plus/pynicotine/downloads.py", line 731, in clear_downloads
  File "D:/a/nicotine-plus/nicotine-plus/pynicotine/downloads.py", line 455, in _clear_transfer
  File "D:/a/nicotine-plus/nicotine-plus/pynicotine/transfers.py", line 348, in _remove_transfer

### Expected behavior
Items should just be cleared/deleted

### Steps to reproduce the bug
Clear all of a user's DLs in the DOWNLOAD tab - presuming it works for ALL statuses

### Additional context
Screenshots, logs, stacktraces or relevant information.

**Comment by slook** [comment]
Thank you for your bug report. Please edit the above post to strip out personal identifier data at `Value:` is is good enough to alter the paste as `'user@@folder\subfolder\filename.ext'` to preserve the peer's privacy, while still providing a useful trace.

**Comment by DeathStalker77** [comment]
@slook  - done!  Sorry about that!  Didn't realize it had that.

**Comment by mathiascode** [comment]
Were you able to reproduce this issue since? I suspect the GUI and core download lists have gone out of sync somehow, but I'm not sure where and why.

**Comment by DeathStalker77** [comment]
I* have not yet tried - currently I am still using the NOV 06 Nightly Build.  I will be switching to the current Nightly Build within the next day, as Win10 will be forcing me to reboot for updates.  I will try doing some queue deletions at that time.  I can say that right now, I have 4 files that were initially stuck at "File Not Shared" (even though they were), and they would NOT change Status - until I tried PAUSE - now I cannot get them "unpaused".

**Comment by DeathStalker77** [comment]
Ok, so when I went to CLOSE the app tonight, I got the following crash error:

Nicotine+ Version: 3.3.0.dev6
GTK Version: 4.12.3
Python Version: 3.11.6 (win32)

Type: <class 'AttributeError'>
Value: 'NoneType' object has no attribute 'rescanning'
Traceback:   File "D:/a/nicotine-plus/nicotine-plus/pynicotine/gtkgui/application.py", line 702, in on_process_thread_events
  File "D:/a/nicotine-plus/nicotine-plus/pynicotine/events.py", line 286, in process_thread_events
  File "D:/a/nicotine-plus/nicotine-plus/pynicotine/events.py", line 245, in emit
  File "D:/a/nicotine-plus/nicotine-plus/pynicotine/events.py", line 328, in _quit
  File "D:/a/nicotine-plus/nicotine-plus/pynicotine/events.py", line 286, in process_thread_events
  File "D:/a/nicotine-plus/nicotine-plus/pynicotine/events.py", line 245, in emit
  File "D:/a/nicotine-plus/nicotine-plus/pynicotine/core.py", line 362, in _thread_callback
  File "D:/a/nicotine-plus/nicotine-plus/pynicotine/uploads.py", line 586, in _check_upload_queue
  File "D:/a/nicotine-plus/nicotine-plus/pynicotine/uploads.py", line 244, in is_new_upload_accepted

**Comment by slook** [comment]
Is it resolved in the latest build?

**Comment by DeathStalker77** [comment]
Do you mean the 18-NOV Build?  I have not yet installed that.

**Comment by slook** [comment]
Yes, https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md#windows

**Comment by DeathStalker77** [comment]
Ok, I will try that one as soon as I'm able to and let you know!  :-)

**Comment by mathiascode** [comment]
There have been some changes since this issue was reported, so I'll tentatively mark it as fixed. Feel free to reopen if you run into the issue again.
